src/cmds/net/arping.c

```diff
@@ -114,15 +113,47 @@ int main(int argc, char **argv) {
 	from.s_addr = in_dev->ifa_address;
 	strncpy(from_b, inet_ntoa(from), sizeof(from_b) - 1);
 	printf("ARPING %s from %s %s\n", dst_b, from_b, in_dev->dev->name);
+
+	struct timeval t1, t2;
+	int ret, microseconds, milliseconds;
+	
 	for (i = 1; i <= cnt; i++) {
 		neighbour_del(ETH_P_IP, &dst, in_dev->dev);
+
+
+		if (gettimeofday(&t1, NULL) == -1)
+					return -1;
+
 		send_request(in_dev->dev, ETH_P_IP, sizeof in_dev->ifa_address,
 				&in_dev->ifa_address, &dst.s_addr);
-		usleep(DEFAULT_INTERVAL);
-		if (0 == neighbour_get_haddr(ETH_P_IP, &dst, in_dev->dev,
-					ARP_HRD_ETHERNET, sizeof hw_addr, &hw_addr[0])) {
+		
+		/* repeat until neigbour_get_haddr returns 0 or one second (default interval) has passed */
+		do  {
+			ret = neighbour_get_haddr(ETH_P_IP, &dst, in_dev->dev,
+								ARP_HRD_ETHERNET, sizeof hw_addr, &hw_addr[0]);
+			if (gettimeofday(&t2, NULL) == -1)
+				return -1;
+
+		} while (ret != 0 && (t2.tv_sec - t1.tv_sec) < 1);
+
+		if (ret == 0 && (t2.tv_sec - t1.tv_sec) < 1){
 			macaddr_print(mac, hw_addr);
-			printf("Unicast reply from %s [%s]  %dms\n", dst_b, mac, 0);
+
+			/* to avoid doing floating point arithmetic */
+			microseconds = (t2.tv_usec - t1.tv_usec);
+			milliseconds = microseconds/1000;
+			microseconds = microseconds % 1000;
+
+
+			printf("Unicast reply from %s [%s]  %d.", dst_b, mac, milliseconds);
+			if (microseconds >= 100)
+				printf("%d", microseconds);
+			else if (microseconds >= 10)
+				printf("0%d", microseconds);
+			else
+				printf("00%d", microseconds);
```

Comment:
There is possible to printf a number with a predefined number of digits with `printf("%03d\n", number)`, you can get a try,

Reply:
After doing that you can get rid of this if-else block and print with a single line of code (i.e. printing milliseconds and microseconds in a one `printf` call).